### DISCUSSION TITLE
Licensing ambiguity

### DISCUSSION THREAD

**Discussion by maciek134** [diasurgical/DevilutionX#4556] [General]
Is this a better place to discuss the license? I don't and won't use Discord, closing issues as spam and sending people there is also pretty rude.

In https://github.com/diasurgical/devilutionX/issues/3338 @AJenbo mentioned that CC licenses are not suitable - that's just a recommendation. No (F)OSS license is going to be appropriate when you have to restrict commercial usage, so you have to be creative.

The current `LICENSE` file in both repos is plainly wrong and contradicts the `README`. You said yourself that the license should be changed in a similar issue: https://github.com/diasurgical/devilution/issues/2208#issuecomment-790689918 - this didn't happen, ~and once someone asked about it the issue was closed like all others.~ (my mistake, the issue is not closed and this just didn't happen yet)

Nobody wants you to allow commercial usage, you can't do that for obvious reasons. You also don't have to use any already existing licenses - most won't work anyway. You are turning potential contributors away by leaving the licensing situation in such an ambiguous state.

**Comment by DakkJaniels** [comment]
Just curious, but why would someone not contribute because of this licensing ambiguity?

**Reply by maciek134** [reply]
The most basic/legal one: I can't fork this project to make a PR, because I don't have the license to modify the code - in my (not a lawyer) opinion the `LICENSE` file is invalid due to the commercial use exception in the `README` and it's safer to treat the code as unlicensed.

As an extension of that, I don't want to put time and effort into a project that doesn't clearly specify users' rights, because who knows how that will look like once the license is finally fixed?

It's just safer to keep away from ambiguous licensing.

**Reply by DakkJaniels** [reply]
🤷🏽‍♂️ Given that Unlicense was applied to the project first, which is an unambiguous waiver of rights, it would seem to me that any text in the `README` is meaningless because there is no way to put those restrictions on the project at this point (I asked a friend, who went to law school, but isn't a practicing lawyer and he agreed with that assessment, but it obviously depends on your jurisdiction). But unless you are trying to do something commercial with this, I don't really see the issue.

**Reply by maciek134** [reply]
If I (and other people who brought it up) didn't see the issue, then this discussion wouldn't be happening.

The complicated part is what happened when the commercial exception was added: are all changes after that not in public domain, since that couldn't have been restricted this way? Are they just without a license because this combination is invalid? That's the thing - I don't want to hire a lawyer before contributing to a project, we have unambiguous licenses for a reason.

**Reply by DakkJaniels** [reply]
Or the readme text means nothing because those rights have already been waived and the original license has not been changed.

**Reply by AJenbo** [reply]
> I don't want to hire a lawyer

Then you probably can't help with resolving this issue.

**Reply by StephenCWills** [reply]
> As an extension of that, I don't want to put time and effort into a project that doesn't clearly specify users' rights...

Perhaps this project is not a good fit for you. I am not a lawyer, but I think it's safe to say that there is literally nothing we can do to guarantee any particular claims in terms of users' rights when they contribute to this project. No amount of clear legal phrasing across any number of documents is going to change the following fact.

> DevilutionX and any of its maintainers are in no way associated with or endorsed by Blizzard Entertainment®.

**Reply by DakkJaniels** [reply]
Also, if you are that concerned about forking, see GitHub's terms of service, more specifically:
5. License Grant to Other Users
Any User-Generated Content you post publicly, including issues, comments, and contributions to other Users' repositories, may be viewed by others. By setting your repositories to be viewed publicly, you agree to allow others to view and "fork" your repositories (this means that others may make their own copies of Content from your repositories in repositories they control).

If you set your pages and repositories to be viewed publicly, you grant each User of GitHub a nonexclusive, worldwide license to use, display, and perform Your Content through the GitHub Service and to reproduce Your Content solely on GitHub as permitted through GitHub's functionality (for example, through forking). You may grant further rights if you [adopt a license](https://docs.github.com/en/articles/adding-a-license-to-a-repository/#including-an-open-source-license-in-your-repository). If you are uploading Content you did not create or own, you are responsible for ensuring that the Content you upload is licensed under terms that grant these permissions to other GitHub Users.

**Comment by AJenbo** [comment]
> Is this a better place to discuss the license?

No not really. License debates tend to get heated, derailed, and lead nowhere. I find that GitHub isn't suited for this type of discussion.

> I don't and won't use Discord

Then you are opting out of fully engaging with the project. Maybe some of the "other people" are willing to do so.

> closing issues as spam and sending people there is also pretty rude.

Don't take it personal "spam" is just the default setting for locking an issue. Writing comments in an already closed issue with clear referrals to take the debate else where could also be seen as rude. Comments like these seem to come in waves from several accounts via multiple channels and are usually not productive. We have also gotten emails and issues about it in related projects today. From my perspective it feels more like people want their voices heard than actually try to help resolve things. You at least had a suggestion though.

> You said yourself that the license should be changed in a similar issue: https://github.com/diasurgical/devilution/issues/2208#issuecomment-790689918 - this didn't happen

That seems like bad faith arguing. The issue isn't marked as resolved so it simply hasn't happened yet.

> Nobody wants you to allow commercial usage, you can't do that for obvious reasons. You also don't have to use any already existing licenses - most won't work anyway. You are turning potential contributors away by leaving the licensing situation in such an ambiguous state.

That may or may not be true, but it gets us no closer to solving the issue.

For the above stated reasons I also intend to lock this discussion from further entries. I don't mean to block you or others from contributing to the project, but I also do not enjoy this type of discussion.

**Reply by maciek134** [reply]
> Then you are opting out of fully engaging with the project.

And you are forcing me to use a service that doesn't respect my privacy and removes transparency from the discussion.

> License debates tend to get heated, derailed, and lead nowhere. I find that GitHub isn't suited for this type of discussion.

With that assumption no place is going to be good for it, and all heat and derailing in this one is coming from people trying to tell me get lost :)

> That seems like bad faith arguing. The issue isn't marked as resolved so it simply hasn't happened yet.

Sorry about that, I confused the linked issue being closed with this one getting closed and locked.

(from comments above)
> Then you probably can't help with resolving this issue.

I mean't that in general, I don't want to hire a lawyer before contributing to projects, not in this specific case (although I don't see how that would help).

Anyway, I appreciate the response and won't bother you again.

**Reply by yuripourre** [reply]
maciek134 I wonder why are you acting like this? But since you mention that you are leaving, I will be happier if I don't get an answer.

**Reply by AJenbo** [reply]
@yuripourre Alright, well this is exactly why we shouldn't have these discussions on GitHub, thread locked.
